### PR TITLE
add event documentation comments;

### PR DESCRIPTION
**Pull request by mjauvin**
- fix wrong argument provided to fireEvent() call ($attr -> $value)
- add event documentation comments

### CODE REVIEW

**Comment by LukeTowers** [comment]
@mjauvin take a look at https://github.com/octobercms/library/blob/master/src/Database/Model.php#L974-L1024. That's what this is supposed to be mirroring. $attr is valid in that one. Specifically notice that these are local events, and actually we can't have the same event key here. We should just change the eloquent model's ones to note that they're also fired on Halcyon models and add a comment as such to the halcyon model events without actually adding a full event docblock.

**Comment by mjauvin** [comment]
Definitely local events, my bad.

And by "We should just change the eloquent model's ones", I assume you mean change the *docblocks* (i.e. add a note to reflect the event is also fired on the Halcyon model, and vice-versa) ?

**Comment by LukeTowers** [comment]
@mjauvin that's correct, change the docblocks to reflect the fact they're also fired on Halcyon models

**Comment by mjauvin** [comment]
Is that acceptable?

File: src/Database/Model.php [hunk 1]
```diff
@@ -1009,6 +1009,8 @@ public function getAttributeValue($key)
          * @event model.getAttribute
          * Called after the model attribute is retrieved
          *
+         * Note: also triggered in October\Rain\Halcyon\Model
```
Comment (LukeTowers) [inline src/Database/Model.php]: Should add this to all events here that are fired in the Halcyon model

Comment (mjauvin) [reply src/Database/Model.php]: Yes, but I wanted to know first if this is how you want it to appear.

Comment (LukeTowers) [reply src/Database/Model.php]: Gotcha. LGTM!

**Comment by daftspunk** [comment]
The fix makes sense. The documentation is a little confusing, why does Halycon need to know about Database and vice versa?

**Comment by mjauvin** [comment]
@daftspunk this is because the API doc will only document one or the other since they both use the same event id. So we added the references

**Comment by daftspunk** [comment]
I see, thanks. I think the documentation was written with the intent to cover only global events. There might be something we can do to namespace the local ones, or include them in the class definitions. This should do for now, though!

**Comment by mjauvin** [comment]
@daftspunk that would be a good thing to add, yes!

**Comment by daftspunk** [comment]
Made a note of it 😉

**Pull request merged**